libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/RoadShieldDownloader.kt

```diff
@@ -21,7 +22,7 @@ internal object RoadShieldDownloader {
 
     suspend fun download(urlToDownload: String): Expected<String, ByteArray> =
         suspendCancellableCoroutine { continuation ->
-            val id = CommonSingletonModuleProvider.createHttpService().request(
+            val id = HttpServiceFactory.getInstance().request(
```

Comment:
We also should reuse that service in line 75, otherwise we can't be sure we properly cancel the request. 
On my device `HttpServiceFactory.getInstance` returns `HttpServiceInterfaceNative`, while `CommonSingletonModuleProvider.httpServiceInstance` returns `MapboxOkHttpService`, so these are definitely different implementations.

Reply:
`HttpServiceInterfaceNative` does ultimately proxy to `MapboxOkHttpService` so there won't be any functional problems but I agree that for clarity we should use the same access point.

Reply:
I verified locally that `CommonSingletonModuleProvider.httpServiceInstance.cancelRequest` does not cancel the request initiated by `HttpServiceFactory.getInstance().request`, but `HttpServiceFactory.getInstance().cancelRequest` does.

Reply:
That was an oversight. I"ll create a new PR for this modification.

Reply:
@Zayankovsky that would have to be a bug. Did you have a chance to set a breakpoint in `MapboxOkHttpService`? You should see the calls being wired there.

Reply:
`MapboxOkHttpService` was indeed involved in `HttpServiceFactory.getInstance()` requests, but it was not the same instance as `CommonSingletonModuleProvider.httpServiceInstance`, and looks like you can't cancel requests using another `MapboxOkHttpService`.

Reply:
This is a bug, looks like with recent deprecation of `CommonSingletonModuleProvider.httpServiceInstance` the ref from the field is not equal to `HttpServiceFactory.getInstance()` anymore. Fixing in https://github.com/mapbox/mapbox-navigation-android/pull/5459.